src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml

```diff
@@ -131,7 +131,8 @@
             <argument type="service" id="serializer.mapping.class_metadata_factory" />
 
             <tag name="api_platform.doctrine.orm.query_extension.item" priority="-8" />
-            <tag name="api_platform.doctrine.orm.query_extension.collection" priority="-8" />
+            <!-- After filter_eager_loading -->
+            <tag name="api_platform.doctrine.orm.query_extension.collection" priority="-18" />
```

Comment:
this priority change breaks filters that rely on existing joins from the eager loading extension because now the eager loading runs after the filters.

Reply:
exmaple filter that now breaks.

```
protected function filterProperty(string $property, $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null)
    {
        if ('searchTerm' !== $property || !in_array($resourceClass, [User::class, Employment::class]))
            return;

        $alias = $rootAlias = $queryBuilder->getRootAliases()[0];
        if (User::class !== $resourceClass) {
            $joins = $queryBuilder->getDQLPart("join");
            if (array_key_exists($rootAlias, $joins)) {
                /** @var \Doctrine\ORM\Query\Expr\Join $join */
                foreach ($joins[$rootAlias] as $join) {
                    if (sprintf('%s.user', $rootAlias) === $join->getJoin())
                        $alias = $join->getAlias();
                }
            }
        }

        $searchParam = $queryNameGenerator->generateParameterName('search');

        $conditions = [];
        $conditions[] = sprintf('%1$s.firstname LIKE :%2$s', $alias, $searchParam);
        $conditions[] = sprintf('%1$s.lastname LIKE :%2$s', $alias, $searchParam);
        $conditions[] = sprintf('CONCAT(%1$s.firstname, \' \', %1$s.lastname) LIKE :%2$s', $alias, $searchParam);
        $conditions[] = sprintf('%1$s.email LIKE :%2$s', $alias, $searchParam);

        $queryBuilder->andWhere(implode(' OR ', $conditions))
            ->setParameter($searchParam, $value . "%");
    }
```

Reply:
we could workaround this issue by adding the join explicitly in the filter, but now we join the relation two times, one by the filter and another by the eager loading extension.

Reply:
> we could workaround this issue by adding the join explicitly in the filter, but now we join the relation two times, one by the filter and another by the eager loading extension.

If the extension works correctly, the existing join is reused then.

@julienfalque do you think it's possible to solve the issue without changing the priority?

Reply:
I'll have to check again, but as I remember this was required to prevent duplicating joins as described in #3487.

Reply:
Just had a look: changing this priority is required to get `EagerLoadingExtension` applied after `FilterEagerLoadingExtension` to fix #3487. I don't think `FilterEagerLoadingExtension`'s priority can be changed (it has to be applied after all filters).

> we could workaround this issue by adding the join explicitly in the filter, but now we join the relation two times, one by the filter and another by the eager loading extension.

If I remember correctly, the changes in this PR make `EagerLoadingExtension` able to rely on existing joins when possible instead of always adding its own. @zolex Are you sure you have duplicated joins if you add one in your filter?

Reply:
I just assumed it because in earlier Versions it happened. I will check the resulting queries tomorrow and let you know.